crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -1,235 +1,519 @@
         }
     }
 }
-
-//TODO: Fill the struct with respective fields
-// Auth Struct
-pub struct DatatransAuthType {
-    pub(super) api_key: Secret<String>,
-}
-
 impl TryFrom<&types::ConnectorAuthType> for DatatransAuthType {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
         match auth_type {
-            types::ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
-                api_key: api_key.to_owned(),
-            }),
+            types::ConnectorAuthType::BodyKey {
+                api_key: passcode,
+                key1: merchant_id,
+            } => {
+                let auth_key = format!("{}:{}", merchant_id.peek(), passcode.peek());
+                let auth_header = format!("Basic {}", consts::BASE64_ENGINE.encode(auth_key));
+                Ok(Self { auth_header })
+            }
             _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
         }
     }
 }
-// PaymentsResponse
-//TODO: Append the remaining status flags
-#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
-#[serde(rename_all = "lowercase")]
-pub enum DatatransPaymentStatus {
-    Succeeded,
-    Failed,
-    #[default]
-    Processing,
-}
 
-impl From<DatatransPaymentStatus> for enums::AttemptStatus {
-    fn from(item: DatatransPaymentStatus) -> Self {
+impl ForeignFrom<(&DatatransResponse, bool)> for enums::AttemptStatus {
+    fn foreign_from((item, is_auto_capture): (&DatatransResponse, bool)) -> Self {
         match item {
-            DatatransPaymentStatus::Succeeded => Self::Charged,
-            DatatransPaymentStatus::Failed => Self::Failure,
-            DatatransPaymentStatus::Processing => Self::Authorizing,
+            DatatransResponse::ErrorResponse(_) => Self::Failure,
+            DatatransResponse::TransactionResponse(_) => {
+                if is_auto_capture {
+                    Self::Charged
+                } else {
+                    Self::Authorized
+                }
+            }
+        }
+    }
+}
+impl From<SyncResponse> for enums::AttemptStatus {
+    fn from(item: SyncResponse) -> Self {
+        match item.res_type {
+            TransactionType::Payment => match item.status {
+                TransactionStatus::Authorized => Self::Authorized,
+                TransactionStatus::Settled | TransactionStatus::Transmitted => Self::Charged,
+                TransactionStatus::Canceled => Self::Voided,
+                TransactionStatus::Failed => Self::Failure,
+                TransactionStatus::Initialized
+                | TransactionStatus::Authenticated
+                | TransactionStatus::ChallengeOngoing
+                | TransactionStatus::ChallengeRequired => Self::Pending,
+            },
+            TransactionType::Credit | TransactionType::CardCheck => Self::Failure,
```

Comment:
If these are unexpected `TransactionTypes` can we remove from the enum itself?

Reply:
Removed | TransactionStatus::ChallengeOngoing
                | TransactionStatus::ChallengeRequired
                But we need TransactionType::Credit | TransactionType::CardCheck